test/Java9andUp/src_latest/org/openj9/test/java/lang/Test_Class.java

```diff
@@ -0,0 +1,63 @@
+package org.openj9.test.java.lang;
+
+import java.lang.Class;
+import java.lang.ClassNotFoundException;
+import java.lang.Module;
+import java.security.AllPermission;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.AssertJUnit;
+
+/*******************************************************************************
+ * Copyright (c) 2017, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+@Test(groups = { "level.extended" })
+@SuppressWarnings("nls")
+public class Test_Class {
+	
+	public static AllPermission		ALL_PERMISSION = new AllPermission();
+
+/**
+ * @tests java.lang.Class#forName(java.lang.String)
+ * @tests java.lang.Class#forName(java.lang.Module, java.lang.String)
+ */
+@Test
+public void test_forName() {
+	try {
+		Class<?> jrAccessExceptionClz = Class.forName("java.rmi.AccessException");
+		AssertJUnit.assertNotNull(jrAccessExceptionClz);
+		if (!jrAccessExceptionClz.getProtectionDomain().implies(ALL_PERMISSION)) {
+			AssertJUnit.fail("java.rmi.AccessException should have all permission!");
+		}
+
+		Module  jrModule = jrAccessExceptionClz.getModule();
+		Class<?> srrRegistryImplClz = Class.forName(jrModule, "sun.rmi.registry.RegistryImpl");
```

Comment:
It would be better to load a class from an exported package to avoid export problems in the future when the default access controls are changed.

Reply:
Will replace it with `java.rmi.AlreadyBoundException` instead.